lib/graphql/fetch/create-fetch.ts

```diff
@@ -0,0 +1,35 @@
+interface CreateFetchOptions {
+  spaceId: string;
+  accessToken: string;
+  previewToken: string;
+}
+
+export function createFetch({
+  spaceId,
+  accessToken,
+  previewToken,
+}: CreateFetchOptions) {
+  async function fetchGraphQL(query: string, variables = {}, preview = false) {
```

Comment:
This can be extended to account for the `tags` option as well to turn this into an `ISR` function and enable the weebhook/revalidation logic handlers